src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/SessionStateTempDataProvider.cs

```diff
@@ -59,16 +59,10 @@ public class SessionStateTempDataProvider : ITempDataProvider
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (!IsSessionEnabled(context))
-            {
-                // Session middleware is not enabled. No-op
-                return null;
-            }
-
             var session = context.Session;
```

Comment:
yes and that is what we want in this scenario. So when the user does `TempData['key1']='value1'`, an exception is thrown to indicate that something has not been configured.